projects/igniteui-angular/src/lib/data-operations/data-util.ts

```diff
@@ -181,60 +180,122 @@ export class DataUtil {
      * @param data Collection to merge
      * @param transactions Transactions to merge into data
      * @param primaryKey Primary key of the collection, if any
+     * @param deleteRows Should delete rows with DELETE transaction type from data
+     * @returns Provided data collections updated with all provided transactions
      */
-    public static mergeTransactions<T>(data: T[], transactions: Transaction[], primaryKey?: any): T[] {
+    public static mergeTransactions<T>(data: T[], transactions: Transaction[], primaryKey?: any, deleteRows: boolean = false): T[] {
         data.forEach((item: any, index: number) => {
             const rowId = primaryKey ? item[primaryKey] : item;
             const transaction = transactions.find(t => t.id === rowId);
-            if (Array.isArray(item.children)) {
-                this.mergeTransactions(item.children, transactions, primaryKey);
-            }
             if (transaction && transaction.type === TransactionType.UPDATE) {
                 data[index] = transaction.newValue;
             }
         });
 
+        if (deleteRows) {
```

Comment:
Does anything besides tests actually use this added delete functionality?

Reply:
Right now no, but we can refactor transaction's `commit` function to call utility's `mergeTransactions` function. As soon as we do this we should call `mergeTransactions` with `deleteRows` set to true

Reply:
Ffair enough